dallinger/experiment.py

```diff
@@ -471,8 +471,7 @@ def run(self, exp_config=None, app_id=None, bot=False, **kwargs):
         """
         import dallinger as dlgr
 
-        if app_id is None:
-            app_id = self.make_uuid()
+        app_id = self.make_uuid(str(app_id))
```

Comment:
So if `app_id` is `None`, `str(app_id)` will return `"None"`, which will fail the `isinstance()` check inside `make_uuid()`, and then we'll generate one?

Reply:
I wrote tests to satisfy my addled brain, so I committed and pushed. Maybe they'll fix the tiny coverage drop.

Reply:
You know, I think moving the str() call into the make_uuid function makes more sense. I was trying to replicate some logic from command_line.py that was clearly put there for some edge case that I didn't want to dig into, but it does confuse the logic and result in an inefficiency.

Reply:
You know, I think moving the str() call into the make_uuid function makes more sense. I was trying to replicate some logic from command_line.py that was clearly put there for some edge case that I didn't want to dig into, but it does confuse the logic and result in an inefficiency.

Thanks for the tests!!